x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -21,89 +21,90 @@ export class JsonIndexFilePicker extends Component {
   state = {
     fileUploadError: '',
     fileParsingProgress: '',
-    fileRef: null
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.fileRef !== this.props.fileRef) {
-      this.setState({ fileRef: this.props.fileRef });
-    }
+  async componentDidMount() {
+    this._isMounted = true;
   }
 
-  _fileHandler = async fileList => {
-    const {
-      resetFileAndIndexSettings, setParsedFile, onFileRemove, onFileUpload,
-      transformDetails, setFileRef, setIndexName
-    } = this.props;
-
-    const { fileRef } = this.state;
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
 
-    resetFileAndIndexSettings();
+  _fileHandler = fileList => {
+    const fileArr = Array.from(fileList);
+    this.props.resetFileAndIndexSettings();
     this.setState({ fileUploadError: '' });
-    if (fileList.length === 0) { // Remove
-      setParsedFile(null);
-      if (onFileRemove) {
-        onFileRemove(fileRef);
-      }
-    } else if (fileList.length === 1) { // Parse & index file
-      const file = fileList[0];
-      if (!file.name) {
+    if (fileArr.length === 0) { // Remove
+      return;
+    }
+    if (fileArr.length === 1) { // Parse & index file
```

Comment:
Is this check needed? Does it matter if more than one file are passed in?

Reply:
I can remove it. It looks like our newer file picker handles multiple files but our current version doesn't(?). We can re-evaluate this type of logic and additional logic for multiple files at a later point